packages/api/src/command/medical/document/document-query-storage-info.ts

```diff
@@ -20,54 +23,56 @@ export type S3Info = {
 };
 
 type SimpleFile = {
-  fhirDocId: string;
   docId: string;
   fileName: string;
   fileLocation: string;
 };
 
-export function docToFile(patient: Pick<Patient, "cxId" | "id">) {
+export function getDocToFileFunction(patient: Pick<Patient, "cxId" | "id">) {
   // TODO convert the input from CW Document to a Metriport shape
-  return (doc: Document): SimpleFile => {
-    const fhirDocId = getDocumentPrimaryId(doc);
-    const fileName = createS3FileName(patient.cxId, patient.id, fhirDocId);
-    return { fhirDocId, docId: doc.id, fileName, fileLocation: s3BucketName };
+  return async (doc: Document): Promise<SimpleFile> => {
+    const extension = getFileExtension(doc.content?.mimeType);
+    const docName = doc.id + (extension ? extension : "");
+    const fileName = createS3FileName(patient.cxId, patient.id, docName);
+    return { docId: doc.id, fileName, fileLocation: s3BucketName };
   };
 }
 
 // TODO convert this to: 1. list files on patient's folder on S3; 2. match to docs and retrieve info
 export async function getS3Info(
-  documents: Document[],
+  documents: DocumentWithMetriportId[],
   patient: Pick<Patient, "cxId" | "id">
 ): Promise<S3Info[]> {
   const { log } = Util.out(`getS3Info - patient ${patient.id}`);
 
-  const errors: { error: unknown; message: string; docId: string; fhirDocId: string }[] = [];
+  const errors: { error: unknown; message: string; docId: string }[] = [];
+  const docToFile = getDocToFileFunction(patient);
```

Comment:
nit: why not just use `return async (doc: Document): Promise<SimpleFile> => {` directly?

Reply:
Do you mean like `convertToNonExistingS3Info`, so the call to `getS3Info()` would move from 

```ts
  const getFilesWithStorageInfo = async () =>
    override
      ? await Promise.all(validDocs.map(convertToNonExistingS3Info(patient)))
      : await getS3Info(validDocs, patient);
```

to

```ts
  const getFilesWithStorageInfo = async () =>
    override
      ? await Promise.all(validDocs.map(convertToNonExistingS3Info(patient)))
      : await Promise.all(validDocs.map(getS3Info(patient));
```

?

I could take a swing at it, but I tried to minimize changes. `getS3Info()` already used `Promise.allSettled()` with code to report on issues.